addons/website_profile/controllers/main.py

```diff
@@ -84,14 +88,16 @@ def get_user_profile_avatar(self, user_id, field='avatar_256', width=0, height=0
             field_name=field, width=int(width), height=int(height), crop=crop
         ).get_response()
 
-    @http.route(['/profile/user/<int:user_id>'], type='http', auth="public", website=True)
+    @http.route(['/profile/user/<int:user_id>'], type='http', auth='user', website=True)
     def view_user_profile(self, user_id, **post):
-        user = self._check_user_profile_access(user_id)
-        if not user:
-            return request.render("website_profile.private_profile")
+        access = self._check_user_profile_access(user_id)
```

Comment:
and so
```suggestion
        user_sudo, denial_reason, user_not_found = self._check_user_profile_access(user_id)
```
It would reduce the number of lines impacted (well except for the added _sudo, but I think it's for the better; or we don't need the sudo and we should `sudo(False)` it somewhere).